coderd/httpmw/taskparam_test.go

```diff
@@ -117,4 +170,89 @@ func TestTaskParam(t *testing.T) {
 		defer res.Body.Close()
 		require.Equal(t, http.StatusOK, res.StatusCode)
 	})
+
+	t.Run("FoundByTaskName", func(t *testing.T) {
+		t.Parallel()
+		rtr := makeRouter()
+		r := makeRequest(user.ID, token)
+		chi.RouteContext(r.Context()).URLParams.Add("task", task.Name)
+		rw := httptest.NewRecorder()
+		rtr.ServeHTTP(rw, r)
+
+		res := rw.Result()
+		defer res.Body.Close()
+		require.Equal(t, http.StatusOK, res.StatusCode)
+	})
+
+	t.Run("NotFoundByWorkspaceName", func(t *testing.T) {
+		t.Parallel()
+		rtr := makeRouter()
+		r := makeRequest(user.ID, token)
+		chi.RouteContext(r.Context()).URLParams.Add("task", workspace.Name)
+		rw := httptest.NewRecorder()
+		rtr.ServeHTTP(rw, r)
+
+		res := rw.Result()
+		defer res.Body.Close()
+		require.Equal(t, http.StatusNotFound, res.StatusCode)
+	})
+
+	t.Run("CaseInsensitiveTaskName", func(t *testing.T) {
+		t.Parallel()
+		rtr := makeRouter()
+		r := makeRequest(user.ID, token)
+		// Look up with different case
+		chi.RouteContext(r.Context()).URLParams.Add("task", "mytask")
+		rw := httptest.NewRecorder()
+		rtr.ServeHTTP(rw, r)
+
+		res := rw.Result()
+		defer res.Body.Close()
+		require.Equal(t, http.StatusOK, res.StatusCode)
+		// Verify we got the right task
+		require.Equal(t, "MyTask", taskMixedCase.Name)
```

Comment:
I believe this is an invalid task name (should be all lower case), I'd suggest reversing the test so that task name is lower and the queried param is mixed case.